# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBaselineCompile.cpp
# Commit: 285174ef42f1
# Full Hash: 285174ef42f184760a177964621271cfe381bbe5
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-23 04:09:13
# Regressor Bug: 1642412
# File Overlap Count: 2
# Description:
#   Bug 1642412 - wasm: Switch ion to use the precise post-write barrier. r=jseward
#   
#   Ion has an easier time switching to the precise post-write barrier
#   as it's already unconditionally doing a call (postBarrierFiltering),
#   and so there's little extra cost in switching that call to be the
# ==============================================================================

diff -r a41351c7413b -r 285174ef42f1 js/src/wasm/WasmBaselineCompile.cpp
--- a/js/src/wasm/WasmBaselineCompile.cpp	Tue Mar 22 19:59:06 2022 +0000
+++ b/js/src/wasm/WasmBaselineCompile.cpp	Tue Mar 22 19:59:07 2022 +0000
@@ -2110,19 +2110,19 @@
 Address BaseCompiler::addressOfTableField(const TableDesc& table,
                                           uint32_t fieldOffset, RegPtr tls) {
   uint32_t tableToTlsOffset =
-      offsetof(TlsData, globalArea) + table.globalDataOffset + fieldOffset;
+      wasm::Instance::offsetOfGlobalArea() + table.globalDataOffset + fieldOffset;
   return Address(tls, tableToTlsOffset);
 }
 
 void BaseCompiler::loadTableLength(const TableDesc& table, RegPtr tls,
                                    RegI32 length) {
-  masm.load32(addressOfTableField(table, offsetof(TableTls, length), tls),
+  masm.load32(addressOfTableField(table, offsetof(TableInstanceData, length), tls),
               length);
 }
 
 void BaseCompiler::loadTableElements(const TableDesc& table, RegPtr tls,
                                      RegPtr elements) {
-  masm.loadPtr(addressOfTableField(table, offsetof(TableTls, elements), tls),
+  masm.loadPtr(addressOfTableField(table, offsetof(TableInstanceData, elements), tls),
                elements);
 }
 
@@ -5906,7 +5906,7 @@
   Label ok;
   masm.wasmBoundsCheck32(
       Assembler::Condition::Below, index,
-      addressOfTableField(table, offsetof(TableTls, length), tls), &ok);
+      addressOfTableField(table, offsetof(TableInstanceData, length), tls), &ok);
   masm.wasmTrap(wasm::Trap::OutOfBounds, bytecodeOffset());
   masm.bind(&ok);
 }